encoding/kzg/verifier/verifier.go

```diff
@@ -18,10 +19,16 @@ import (
 	"github.com/consensys/gnark-crypto/ecc"
 	"github.com/consensys/gnark-crypto/ecc/bn254"
 	"github.com/consensys/gnark-crypto/ecc/bn254/fr"
+	_ "go.uber.org/automaxprocs"
 )
 
+// VerifierOption defines a function that configures a Verifier
+type VerifierOption func(*Verifier) error
```

Comment:
This can go in it's own PR, it's applying the same options pattern refactor to verifier so not super relevant to GPU encoder

Reply:
It looks more verbose and is deviating from the convention, what're the main problems it solves here compared to passing in a config struct?

Reply:
One argument I would give for this configuration pattern is that it provides a simpler constructor that is extensible and backwards compatible. 

For example, with a config struct we would have to modify every client that calls the constructor when we add a new configuration value vs. this pattern you add a `WithX` function in the library that sets the field and does whatever validation it needs to. Then clients can optionally update to use that new added configuration.

Reply:
>with a config struct we would have to modify every client that calls the constructor when we add a new configuration value

Not sure about this, if that new field is optional, the caller doesn't need to do anything. If it's required, callers will have to make changes in either case.

My concerns are the boilplate and the scalabilty of keeping adding WithXX, we may have a long list of fields (like in some big structs).
Also it looks it still need to operate on config struct, if the validation runs across multiple fields (eg. invariants defined over multiple fields).

Reply:
> My concerns are the boilplate and the scalabilty of keeping adding WithXX, we may have a long list of fields (like in some big structs).

These seem tractable problem though, here is an example in grpc library:
https://github.com/grpc/grpc-go/blob/66385b28b3fe71a4895f00d581ede0a344743a3f/dialoptions.go#L248

Reply:
Think I agree with both of you. I've grown fond of the options pattern and like it best, but I still think consistency across a codebase is the most important thing, so kind of agree with jian about this.

Reply:
No problem, I'll revert to the previous method